File: backend/app/schemas/generated_video.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import VideoStatus


class GeneratedVideoResponse(BaseModel):
    id: uuid.UUID
    video_session_id: uuid.UUID
    version: int
    status: VideoStatus
    progress_percent: int
    current_scene: int | None
    file_url: str | None
    error_message: str | None
    created_at: datetime
    completed_at: datetime | None

    model_config = {"from_attributes": True}


class GeneratedVideoListResponse(BaseModel):
    videos: list[GeneratedVideoResponse]
    any_active: bool  # True when any video is QUEUED or RENDERING — drives frontend polling

File: frontend/src/types/generated-video.ts
export type VideoStatus = "queued" | "rendering" | "ready" | "failed";

export interface GeneratedVideo {
  id: string;
  video_session_id: string;
  version: number;
  status: VideoStatus;
  progress_percent: number;
  current_scene: number | null;
  file_url: string | null;
  error_message: string | null;
  created_at: string;
  completed_at: string | null;
}

export interface GeneratedVideoListResponse {
  videos: GeneratedVideo[];
  any_active: boolean;
}
